src/Tools/dotnet-monitor/CollectionRules/Triggers/EventPipeTriggerFactory.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.EventPipe;
+using Microsoft.Diagnostics.Monitoring.EventPipe.Triggers;
+using Microsoft.Diagnostics.Monitoring.EventPipe.Triggers.Pipelines;
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.NETCore.Client;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Triggers
+{
+    internal sealed class EventPipeTriggerFactory
+    {
+        /// <summary>
+        /// Creates a collection rule trigger sourced from the event pipe of the target
+        /// process represented by the specified endpoint.
+        /// </summary>
+        public ICollectionRuleTrigger Create<TSettings>(
```

Comment:
So we should be able to use this factory for all EventPipe based triggers correct?

Reply:
Correct